docs/design-guides/architecture-overview.md

```diff
@@ -0,0 +1,49 @@
+- There is Graph open extensions rate limit of 455 requests per 10 seconds as suggested [here](https://docs.microsoft.com/en-us/graph/throttling#open-and-schema-extensions-service-limits)
+- The sample solution leveraging Graph open extensions might not be suitable for the scenarios where the Azure Active Directory instance is used for 3rd party application sign in as with [delegated permissions granted over Graph Api](https://docs.microsoft.com/en-us/graph/auth/auth-concepts#delegated-and-application-permissions). The 3rd Party application in this case with right delegated permissions would be able to access the open extension data of the user.
+> Note: From data privacy perspective, the ACS Identity is categorized as [EUPI data](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/properties/devenv-dataclassification-property#property-value)
+
+## Guidance
+1. To support the multi environment scenario within Graph Open Extensions using the same Azure Active Directory user instance, you could extend the [IdentityMappingModel class](https://github.com/Azure-Samples/communication-services-authentication-hero-csharp/blob/main/src/Models/IdentityMappingModel.cs) to handle multiple environments.
+2. If Graph open extensions does not meet your requirement for storage of Identity Mappings, you can consider storing them in databases as [CosmoDB](https://docs.microsoft.com/en-us/azure/cosmos-db/) or [Azure Tables](https://docs.microsoft.com/en-us/azure/storage/tables/) for an example. And you would need to update the write and read operations in [GraphService.cs class](https://github.com/Azure-Samples/communication-services-authentication-hero-csharp/blob/main/src/Services/GraphService.cs)
+    - Replace the usage of Graph `/me/extensions` endpoint with [`/me`](https://docs.microsoft.com/en-us/graph/api/resources/users?view=graph-rest-1.0) endpoint. The `/me` endpoint will get the M365 user id in response as `id` attribute. This could be used as key to map the ACS Identity. The permissions for Graph Api on server app registrations just needs to be reduced to `User.Read`.
+    - Persist the mapping of ACS Identity to M365 user id within the database of your choice.
+    - Since the ACS Identity is classified as EUPI, please make sure all the required data privacy compliance are met on your end while the data is at rest and in transit.
+3. Since Azure Communication Services is a data processor and you are the controller of the user data, you are responsible for ensuring the data privacy complaince. Please visit [here](https://docs.microsoft.com/en-us/azure/communication-services/concepts/privacy) for more information.
```

Comment:
This feels like us transferring our internal terminology/complexity/frame of reference onto the user.

Reply:
Regarding compliance , I believe all companies should already be aware of how they need to make the data compliant , we just need to make sure to provide necessary handles to them. As in our sample case the Delete user endpoint is something to help our customers meet compliance. This was something Haider had provided feedback sometime back. 
Nevertheless I think this point may still need to rephrased well and may be needs more clarifications for users to make them actionable. I again plan to pick this up when Haider is back this week.

@ddematheu2 would you have additional thoughts here.